spec/utils/requests_spec.rb

```diff
@@ -12,12 +12,26 @@
   let(:http) { Net::HTTP.new('test') }
 
   describe '#get_response' do
-    it 'call Net::HTTP.get' do
-      allow(Net::HTTP).to receive(:start).and_yield(http)
-      Net::HTTP.start do |http|
-        expect(http).to receive(:get)
+    context 'Valid URL is given' do
+      let(:uri) { 'http://example.com' }
+      it 'call Net::HTTP.get' do
+        allow(Net::HTTP).to receive(:start).and_yield(http)
+        Net::HTTP.start do |http|
+          expect(http).to receive(:get)
+        end
+        subject.get_response(uri, device, open_timeout, read_timeout)
+      end
+    end
+    context 'Invalid URI is given' do
+      ['https://example.com/path/query[]=1/', 'http://例.com?query=[]'].each do |uri|
```

Comment:
I think these URLs are also 'valid' URLs.
so you have to use 'unusual' or 'rare'.

Otherwise, I think putting together is better.
```
it 'call Net::HTTP.get' do
  ['http://example.com', 'https://example.com/path/query[]=1/', 'http://例.com?query=[]'].each do |uri|

  end
end
```

Reply:
Fixed.